catalog/rest/rest.go

```diff
@@ -750,27 +750,12 @@ func (r *Catalog) CreateTable(ctx context.Context, identifier table.Identifier,
 		o(&cfg)
 	}
 
-	freshSchema, err := iceberg.AssignFreshSchemaIDs(schema, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	freshPartitionSpec, err := iceberg.AssignFreshPartitionSpecIDs(cfg.PartitionSpec, schema, freshSchema)
-	if err != nil {
-		return nil, err
-	}
-
-	freshSortOrder, err := table.AssignFreshSortOrderIDs(cfg.SortOrder, schema, freshSchema)
-	if err != nil {
-		return nil, err
-	}
```

Comment:
neither java, nor rust are doing this, I'd prefer to get rid of it to remove the last callsite of `iceberg.AssignFreshPartitionSpecIDs`. Any rest catalog needs to recheck this on createTable, servers can't trust clients to do this beforehand so I think this is largely redundant

Reply:
It makes very little sense for view or namespace changes to fail based on these changes?

```
    --- PASS: TestRestIntegration/TestCreateNamespace (0.03s)
    --- FAIL: TestRestIntegration/TestCreateTable (0.02s)
    --- FAIL: TestRestIntegration/TestCreateView (0.01s)
    --- PASS: TestRestIntegration/TestLoadNamespaceProps (0.01s)
    --- PASS: TestRestIntegration/TestUpdateNamespaceProps (0.03s)
    --- FAIL: TestRestIntegration/TestWriteCommitTable (0.02s)
```

maybe test setup is already failing?

Reply:
Yea, just confirmed that current main passes without issue. Most likely the fact that we're not re-assigning the field/partition/schema IDs is getting rejected by the iceberg rest image

Reply:
will investigate a bit further, worst case, I'll just put the reassignments back in

Reply:
Looks like the failure is caused by `"write-order":{"order-id": 0, "fields": null}` instead of `"fields": []`. So something about the json marshalling of the payload without the Assign functions is leaving a slice as `null` instead of an empty slice. Let me take a look

Reply:
Yeah, it's just the null value of `var cfg catalog.CreateTableCfg` ruining the day

Reply:
yea, the default in `CreateTableCfg` is the zero value which has `nil` for the slice instead of an empty slice. The rest-fixture errors on the `null`.

You can fix it by adding:

```go
if cfg.SortOrder.Fields == nil {
    cfg.SortOrder = table.UnsortedSortOrder
}
```

before the payload is created

Reply:
eventually we can shift to jsonv2 which will use an empty slice as the default instead of nil

Reply:
I was about to add a constructor which would put sane defaults?


```
func NewCreateTableCfg() CreateTableCfg {
	return CreateTableCfg{
		 Location:      "",
		 PartitionSpec: nil,
		 SortOrder:     table.UnsortedSortOrder,
		 Properties:    nil,
	}
}
```